c_sharp_json_diff/JsonDiff.cs

```diff
@@ -1,55 +1,101 @@
+        {
+            foreach (KeyValuePair<string, JToken?> prop in jObject)
             {
-                first.TryGetValue(key, out double firstValue);
-                second.TryGetValue(key, out double secondValue);
-                if(firstValue != secondValue)
+                if(prop.Key == "_t")
                 {
-                    Diff item = new Diff();
-                    item.jsonProperty = key;
-                    item.previous = firstValue;
-                    item.current = secondValue;
-                    item.difference = Math.Abs(firstValue - secondValue);
-                    result.Diffs.Add(item);
+                    continue;
+                }
+                if (prop.Value != null)
+                {
+                    if (prop.Value is JArray)
+                    {
+                        JArray valuesAsArray = (JArray)prop.Value;
+                        if (valuesAsArray[0] != null && valuesAsArray[1] != null)
+                        {
+                            string leftValueAsString = valuesAsArray[0].ToString();
+                            string rightValueAsString = valuesAsArray[1].ToString();
+                            if (int.TryParse(leftValueAsString, out int leftValueAsInt) && int.TryParse(rightValueAsString, out int rightValueAsInt))
+                            {
+                                int diffValue = isAbsoluteDiff ? Math.Abs(leftValueAsInt - rightValueAsInt) : leftValueAsInt - rightValueAsInt;
+                                Dictionary<string, int> map = new Dictionary<string, int>();
+                                map.Add("left", leftValueAsInt);
+                                map.Add("right", rightValueAsInt);
+                                map.Add("diff", diffValue);
+                                valuesAsArray.RemoveAll();
+                                valuesAsArray.Add(JToken.FromObject(map));
+                            }
+                            else if (float.TryParse(leftValueAsString, out float leftValueAsFloat) && float.TryParse(rightValueAsString, out float rightValueAsFloat))
+                            {
+                                float diffValue = isAbsoluteDiff ? Math.Abs(leftValueAsFloat - rightValueAsFloat) : leftValueAsFloat - rightValueAsFloat;
+                                Dictionary<string, float> map = new Dictionary<string, float>();
+                                map.Add("left", leftValueAsFloat);
+                                map.Add("right", rightValueAsFloat);
+                                map.Add("diff", diffValue);
+                                valuesAsArray.RemoveAll();
+                                valuesAsArray.Add(JToken.FromObject(map));
+                            }
+
+                        }
+                    }
+                    else
+                    {
+                        CalculateNumericDiff((JObject)prop.Value, isAbsoluteDiff);
```

Comment:
Is this cast safe? Is it always going to be a `JObject`?

Reply:
I think there are cases where it wouldn't be JObject but I never came across those issues. I added the if block to make casting safe